From: Alexei Starovoitov <ast@kernel.org>
Date: Tue, 14 Jul 2020 12:31:45 -0700
Subject: bpfilter: Initialize pos variable
Patch-mainline: v5.9-rc1
Git-commit: a4fa458950b40d3849946daa32466392811a3716
References: bsc#1155518 bsc#1177028

Make sure 'pos' is initialized to zero before calling kernel_write().

Fixes: d2ba09c17a06 ("net: add skeleton of bpfilter kernel module")
Signed-off-by: Alexei Starovoitov <ast@kernel.org>
Acked-by: Gary Lin <glin@suse.com>
---
 net/bpfilter/bpfilter_kern.c |    1 +
 1 file changed, 1 insertion(+)

--- a/net/bpfilter/bpfilter_kern.c
+++ b/net/bpfilter/bpfilter_kern.c
@@ -47,6 +47,7 @@ static int __bpfilter_process_sockopt(st
 	req.len = optlen;
 	if (!bpfilter_ops.info.tgid)
 		goto out;
+	pos = 0;
 	n = kernel_write(bpfilter_ops.info.pipe_to_umh, &req, sizeof(req),
 			   &pos);
 	if (n != sizeof(req)) {
